src/Wallabag/CoreBundle/Controller/EntryController.php

```diff
@@ -252,7 +253,13 @@ private function showEntries($type, Request $request, $page)
         $entries = new Pagerfanta($pagerAdapter);
 
         $entries->setMaxPerPage($this->getUser()->getConfig()->getItemsPerPage());
-        $entries->setCurrentPage($page);
+        try {
+            $entries->setCurrentPage($page);
+        } catch (OutOfRangeCurrentPageException $e) {
+            if ($page > 1) {
+                return $this->redirect($this->generateUrl($type, array('page' => $entries->getNbPages())), 301);
```

Comment:
Also, could you put a `302` instead? You don't know if in the future you'll have the given page :)